src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -169,6 +169,12 @@ public void finish() {
      */
     @Override
     public int deflate(byte[] b, int off, int len ) {
```

Comment:
Can you please check if we need `off` ?

Reply:
https://github.com/Intel-HLS/GKL/blob/master/src/main/java/com/intel/gkl/compression/IntelDeflater.java  has the API.  

Off is offset from the byte pointer.  It seems normal type thing to me.   Need to be sure off + len is not greater then sizeof b.

Reply:
I am trying to understand where it's assigned. It should be either [declared](https://github.com/Intel-HLS/GKL/blob/master/src/main/java/com/intel/gkl/compression/IntelDeflater.java#L121) and passed with java object as inputBufferOffset in the structure, however it is commneted in counterpart
[here](https://github.com/Intel-HLS/GKL/blob/08747449e8fc9d42e4ac270ef5077ada9d19ed0a/src/main/native/compression/IntelDeflater.cc#L198)

Reply:
Suggestion to cover this validation with `if(off < 0 || off > b.length - len) `

Reply:
Here `off` is dummy input can simply check for null and assign as 0; than throwing exception.